opentreemap/treemap/decorators.py

```diff
@@ -130,6 +130,20 @@ def wrapped(request, instance, *args, **kwargs):
     return wrapper_function
 
 
+def requires_permission(codename):
+    def wrapper_function(view_fn):
+        @wraps(view_fn)
+        def wrapped(request, instance, *args, **kwargs):
+            if request.instance_user.role.has_permission(codename):
```

Comment:
Permissions live on roles, and only an instance user has a role. So it only makes sense to use `requires_permission` inside `instance_request`, which sets `request.instance_user`. That said, I've added some safety code.